engine/api/api.go

```diff
@@ -583,26 +583,27 @@ func (a *API) Serve(ctx context.Context) error {
 		log.Error("error while initializing workers routine: %s", err)
 	}
 
-	log.Info("Initializing internal routines...")
-	go queue.Pipelines(ctx, a.Cache, a.DBConnectionFactory.GetDBMap)
-	go pipeline.AWOLPipelineKiller(ctx, a.DBConnectionFactory.GetDBMap, a.Cache)
-	go hatchery.Heartbeat(ctx, a.DBConnectionFactory.GetDBMap)
-	go auditCleanerRoutine(ctx, a.DBConnectionFactory.GetDBMap)
-	go metrics.Initialize(ctx, a.DBConnectionFactory.GetDBMap, a.Config.Name)
-	go repositoriesmanager.ReceiveEvents(ctx, a.DBConnectionFactory.GetDBMap, a.Cache)
-	go action.RequirementsCacheLoader(ctx, 5*time.Second, a.DBConnectionFactory.GetDBMap, a.Cache)
-	go hookRecoverer(ctx, a.DBConnectionFactory.GetDBMap, a.Cache)
-	go services.KillDeadServices(ctx, services.NewRepository(a.mustDB, a.Cache))
-	go poller.Initialize(ctx, a.Cache, 10, a.DBConnectionFactory.GetDBMap)
-	go migrate.CleanOldWorkflow(ctx, a.Cache, a.DBConnectionFactory.GetDBMap, a.Config.URL.API)
-	go migrate.KeyMigration(a.Cache, a.DBConnectionFactory.GetDBMap, &sdk.User{Admin: true})
-	go broadcast.Initialize(ctx, a.DBConnectionFactory.GetDBMap)
-
 	a.warnChan = make(chan sdk.Event)
 	event.Subscribe(a.warnChan)
-	go workflow.ComputeAudit(ctx, a.DBConnectionFactory.GetDBMap)
-	go warning.Start(ctx, a.DBConnectionFactory.GetDBMap, a.warnChan)
-	go a.serviceAPIHeartbeat(ctx)
+
+	log.Info("Initializing internal routines...")
+	sdk.GoRoutine("workflow.ComputeAudit", func() { workflow.ComputeAudit(ctx, a.DBConnectionFactory.GetDBMap) })
+	sdk.GoRoutine("warning.Start", func() { warning.Start(ctx, a.DBConnectionFactory.GetDBMap, a.warnChan) })
+	sdk.GoRoutine("queue.Pipelines", func() { queue.Pipelines(ctx, a.Cache, a.DBConnectionFactory.GetDBMap) })
+	sdk.GoRoutine("pipeline.AWOLPipelineKiller", func() { pipeline.AWOLPipelineKiller(ctx, a.DBConnectionFactory.GetDBMap, a.Cache) })
+	sdk.GoRoutine("hatchery.Heartbeat", func() { hatchery.Heartbeat(ctx, a.DBConnectionFactory.GetDBMap) })
+	sdk.GoRoutine("auditCleanerRoutine(ctx", func() { auditCleanerRoutine(ctx, a.DBConnectionFactory.GetDBMap) })
+	sdk.GoRoutine("metrics.Initialize", func() { metrics.Initialize(ctx, a.DBConnectionFactory.GetDBMap, a.Config.Name) })
+	sdk.GoRoutine("repositoriesmanager.ReceiveEvents", func() { repositoriesmanager.ReceiveEvents(ctx, a.DBConnectionFactory.GetDBMap, a.Cache) })
+	sdk.GoRoutine("action.RequirementsCacheLoader", func() { action.RequirementsCacheLoader(ctx, 5*time.Second, a.DBConnectionFactory.GetDBMap, a.Cache) })
+	sdk.GoRoutine("hookRecoverer(ctx", func() { hookRecoverer(ctx, a.DBConnectionFactory.GetDBMap, a.Cache) })
+	sdk.GoRoutine("services.KillDeadServices", func() { services.KillDeadServices(ctx, services.NewRepository(a.mustDB, a.Cache)) })
```

Comment:
Error return value of `services.KillDeadServices` is not checked

Reply:
fixed